components/google-cloud/tests/aiplatform/test_componets_compile.py

```diff
@@ -281,25 +272,58 @@ def pipeline():
                 project=self._project,
                 display_name=self._display_name,
                 serving_container_image_uri=self._serving_container_image_uri,
-                artifact_uri=self._artifact_uri
-            )
+                artifact_uri=self._artifact_uri)
 
             endpoint_create_op = EndpointCreateOp(
-                project=self._project, display_name=self._display_name
-            )
+                project=self._project, display_name=self._display_name)
 
             model_deploy_op = ModelDeployOp(
-                project=self._project, model=model_upload_op.outputs["model"]
-            )
+                project=self._project, model=model_upload_op.outputs["model"])
 
             batch_predict_op = ModelBatchPredictOp(
                 project=self._project,
                 model=model_upload_op.outputs["model"],
                 job_display_name=self._display_name,
                 gcs_source=self._gcs_source,
-                gcs_destination_prefix=self._gcs_destination_prefix
-            )
+                gcs_destination_prefix=self._gcs_destination_prefix)
+
+        compiler.Compiler().compile(
+            pipeline_func=pipeline, package_path=self._package_path)
+
+    def test_model_upload_op_compile(self):
+
+        @kfp.dsl.pipeline(name="training-test")
```

Comment:
+1 i see your point on validating the IR. Should we distinguish these tests from unit tests since they are actually running E2E integration between kfp and gcpc ?  currently we have the tests under a separate folder , we could move all tests under a folder for each component similar to tensorflow cloud [see](https://github.com/tensorflow/cloud/tree/master/src/python/tensorflow_cloud/core/tests) then we can separate unit test and integration tests with a similar folder structure to tf_cloud. this could be a post release clean up task.

Reply:
just to clarify this is still  a unit test. the test is intend to validate the YAML generates the correct IR. There are concat logics goes into the YAML to construct payload, and there is a risk of bugs in there so the payload is not in a right json format. this test is intend to capture that (by providing values for all parameters)

Reply:
got it, that makes sense.

Reply:
i saw your point. it's indeed an integration test. I'll move it to a separate folder.